### PR TITLE
Added all local data sources for v6.

### PR DESCRIPTION
**Pull request by Turhsus**
Please take a look!

### CODE REVIEW

File: pkg/artifactory/datasource/repository/local/datasource_artifactory_local_cargo_repository.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	"github.com/jfrog/terraform-provider-artifactory/v7/pkg/artifactory/resource/repository/local"
+	"github.com/jfrog/terraform-provider-shared/packer"
+)
+
+func DataSourceArtifactoryLocalCargoRepository() *schema.Resource {
+	constructor := func() (interface{}, error) {
+		return &local.CargoLocalRepoParams{
+			RepositoryBaseParams: local.RepositoryBaseParams{
+				PackageType: "cargo",
+				Rclass:      "local",
```
Comment (alexhung) [inline pkg/artifactory/datasource/repository/local/datasource_artifactory_local_cargo_repository.go:14]: Use `rclass`

File: pkg/artifactory/datasource/repository/local/datasource_artifactory_local_debian_repository.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	"github.com/jfrog/terraform-provider-artifactory/v7/pkg/artifactory/resource/repository/local"
+	"github.com/jfrog/terraform-provider-shared/packer"
+)
+
+func DataSourceArtifactoryLocalDebianRepository() *schema.Resource {
+	constructor := func() (interface{}, error) {
+		return &local.DebianLocalRepositoryParams{
+			RepositoryBaseParams: local.RepositoryBaseParams{
+				PackageType: "debian",
+				Rclass:      "local",
```
Comment (alexhung) [inline pkg/artifactory/datasource/repository/local/datasource_artifactory_local_debian_repository.go:14]: Same here

**Review by alexhung — CHANGES_REQUESTED** [review]
Minor things. Everything else look good.

File: pkg/artifactory/datasource/repository/local/datasource_artifactory_local_java_repository.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+)
+
+func DataSourceArtifactoryLocalJavaRepository(repoType string, suppressPom bool) *schema.Resource {
+	javaLocalSchema := local.GetJavaRepoSchema(repoType, suppressPom)
+
+	constructor := func() (interface{}, error) {
+		return &local.JavaLocalRepositoryParams{
+			RepositoryBaseParams: local.RepositoryBaseParams{
+				PackageType: repoType,
+				Rclass:      "local",
```
Comment (alexhung) [inline pkg/artifactory/datasource/repository/local/datasource_artifactory_local_java_repository.go:16]: `rclass` here too. Please check the rest of the PR.

**Review by alexhung — CHANGES_REQUESTED** [review]

**Review by alexhung — APPROVED** [review]

**Review by alexhung — COMMENTED** [review]
Looks good. Just need the doc files for the missing local repo types.

**Comment by danielmkn** [comment]
@alexhung what is missing in the documentation? I don't see missing docs.

**Comment by alexhung** [comment]
@danielmkn Many package types (Bower, etc.) are missing dedicated documentation files. They are generic types but we don't call out that those package types should refers to the generic doc. Either we add separate file for each of these 'generic' package types, or include some text in the `local.md`. Currently users have no easy way to know which package types are supported or not because some have doc files, some don't.

**Comment by danielmkn** [comment]
Oh, then we need to add one .md file per each type, the same as we have for the resources. Thoughts?

**Comment by alexhung** [comment]
@danielmkn Yes, exactly. We have files for each local, remote, virtual, and federated repo package type resources so I think we need the same documentation for data source.

File: docs/data-sources/local_bower_repository.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+---
+
+# Artifactory Local Bower Repository Data Source
+
+Retrieves a local Bower repository.
+
+## Example Usage
+
+```hcl
+data "artifactory_local_bower_repository" "terraform-local-test-bower-repo" {
```
Comment (alexhung) [inline docs/data-sources/local_bower_repository.md:12]: `terraform`? 😄

**Review by alexhung — CHANGES_REQUESTED** [review]
🚢

File: docs/data-sources/local_bower_repository.md (hunk 4 above)
Comment (Turhsus) [reply docs/data-sources/local_bower_repository.md:12]: Oh no its copy pasted down the list. Will remove

**Review by danielmkn — APPROVED** [review]

**Pull request merged**